src/app/service/change-password.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { Entity } from '../entity/entity';
+import {Http,Headers} from '@angular/http';
+import 'rxjs/add/operator/map';
+declare const require: any;
+const CryptoJS = require('crypto-js');
+@Injectable()
+export class ChangePasswordService {
+
+  constructor() { }
+
+  public name: string;
+  public lastUsedVersion: string;
+  private password: any;
+
+  public getPrefix(): string {
+    return 'user:';
+  }
+
+  public setNewPassword(password) {
+    const cryptKeySize = 256 / 32;
+    const cryptIterations = 128;
+    const cryptSalt = CryptoJS.lib.WordArray.random(128 / 8).toString();
+    const hash = CryptoJS.PBKDF2(password, cryptSalt, {
+      keySize: cryptKeySize,
+      iterations: cryptIterations
+    }).toString();
+    console.log('a');
```

Comment:
Please remove any debug functions from the code before a PR. (best to exclude such lines when making a commit)